spec/rails_helper.rb

```diff
@@ -0,0 +1,63 @@
+# This file is copied to spec/ when you run 'rails generate rspec:install'
+require 'spec_helper'
+ENV['RAILS_ENV'] ||= 'test'
+require_relative '../config/environment'
+# Prevent database truncation if the environment is production
+abort('The Rails environment is running in production mode!') if Rails.env.production?
+require 'rspec/rails'
+# Add additional requires below this line. Rails is not loaded until this point!
+
+# Requires supporting ruby files with custom matchers and macros, etc, in
+# spec/support/ and its subdirectories. Files matching `spec/**/*_spec.rb` are
+# run as spec files by default. This means that files in spec/support that end
+# in _spec.rb will both be required and run as specs, causing the specs to be
+# run twice. It is recommended that you do not name files matching this glob to
+# end with _spec.rb. You can configure this pattern with the --pattern
+# option on the command line or in ~/.rspec, .rspec or `.rspec-local`.
+#
+# The following line is provided for convenience purposes. It has the downside
+# of increasing the boot-up time by auto-requiring all files in the support
+# directory. Alternatively, in the individual `*_spec.rb` files, manually
+# require only the support files necessary.
+#
+# Dir[Rails.root.join('spec', 'support', '**', '*.rb')].sort.each { |f| require f }
+
+# Checks for pending migrations and applies them before tests are run.
+# If you are not using ActiveRecord, you can remove these lines.
+begin
+  ActiveRecord::Migration.maintain_test_schema!
+rescue ActiveRecord::PendingMigrationError => e
+  abort e.to_s.strip
+end
+RSpec.configure do |config|
+  # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
+  config.fixture_path = "#{:Rails.root}/spec/fixtures"
```

Comment:
Currently, A noMethodError which comes from the rails_helper is thrown when you try to run your tests. The error message indicates that :Rails.root is being treated as a symbol (:Rails) and the method root is not defined for symbols😉. Kindly see the suggestion below😉.

```suggestion
  config.fixture_path = "#{Rails.root}/spec/fixtures"
```

When you do this your test should run properly. but kindly note that some of your tests are still failing😉. Kindly ensue all your tests are passing to avoid blockers as this is a requirement.


<img width="541" alt="image" src="https://github.com/BilalLiaquat7/blog_app/assets/64643995/32a819f5-3653-4409-b98c-0c77c2b5086a">